lib/std/time.zig

```diff
@@ -59,6 +59,398 @@ pub fn sleep(nanoseconds: u64) void {
+    year: i32,
+
+    /// UNIX timestamp in nanoseconds
+    nano_timestamp: i128,
+
+    /// The time zone to which `nano_timestamp` was projected to.
+    tz_projection: TimeZone.Projection,
+
+    pub const Weekday = enum(u3) {
+        monday = 1,
+        tuesday = 2,
+        wednesday = 3,
+        thursday = 4,
+        friday = 5,
+        saturday = 6,
+        sunday = 7,
+
+        pub const names = [_][]const u8{
+            "Monday",
+            "Tuesday",
+            "Wednesday",
+            "Thursday",
+            "Friday",
+            "Saturday",
+            "Sunday",
+        };
+
+        pub fn shortName(w: Weekday) []const u8 {
+            return names[@intFromEnum(w) - 1][0..3];
+        }
+
+        pub fn longName(w: Weekday) []const u8 {
+            return names[@intFromEnum(w) - 1];
+        }
+    };
+
+    pub const Month = enum(u4) {
+        january = 1,
+        february = 2,
+        march = 3,
+        april = 4,
+        may = 5,
+        june = 6,
+        july = 7,
+        august = 8,
+        september = 9,
+        october = 10,
+        november = 11,
+        december = 12,
+
+        pub const names = [_][]const u8{
+            "January",
+            "February",
+            "March",
+            "April",
+            "May",
+            "June",
+            "July",
+            "August",
+            "September",
+            "October",
+            "November",
+            "December",
+        };
+
+        pub fn shortName(m: Month) []const u8 {
+            return names[@intFromEnum(m) - 1][0..3];
+        }
+
+        pub fn longName(m: Month) []const u8 {
+            return names[@intFromEnum(m) - 1];
+        }
+    };
+
+    /// Get current date and time in the system's local time.
+    /// On non-Windows systems this loads the system's local time
+    /// time zone database from the filesystem and may allocate
+    /// so when called repeated `nowTz` should be preferred.
+    pub fn now(allocator: std.mem.Allocator) LocalTimeError!DateTime {
+        const nano_timestamp = nanoTimestamp();
+
+        if (builtin.os.tag == .windows) {
+            var tzi: std.os.windows.TIME_ZONE_INFORMATION = undefined;
+            const rc = std.os.windows.ntdll.NtQuerySystemInformation(
+                .SystemCurrentTimeZoneInformation,
+                &tzi,
+                @sizeOf(std.os.windows.TIME_ZONE_INFORMATION),
+                null,
+            );
+            if (rc != .SUCCESS) return error.LocalTimeUnavailable;
+            return fromTimestamp(nano_timestamp, .{
+                .name_data = .{
+                    @intCast(tzi.StandardName[0]),
+                    @intCast(tzi.StandardName[1]),
+                    @intCast(tzi.StandardName[2]),
+                    @intCast(tzi.StandardName[3]),
+                    @intCast(tzi.StandardName[4]),
+                    @intCast(tzi.StandardName[5]),
+                },
+                .offset = tzi.Bias,
```

Comment:
[The documentation](https://learn.microsoft.com/en-us/windows/win32/api/timezoneapi/ns-timezoneapi-time_zone_information) doesn't mention if this includes leap seconds but I'd guess it does?

Reply:
Not sure. If not, there is https://github.com/microsoft/STL/discussions/1624 and https://www.codeproject.com/Articles/8789/World-Clock-and-the-TimeZoneInformation-class on how to do it.

Bear in mind, that Linux may or may not be configured to use UTC or TAI directly without (from my understanding) a standard way to check the current config ie for folks hosting their own UTC servers (without standard query interface) see https://en.wikipedia.org/wiki/Unix_time#Variant_that_counts_leap_seconds.